### PR TITLE
Fix /dt showing 'undefined' as Zygarde ability

### PR DESCRIPTION
**Pull request by asgdf**
Special case for Zygarde since it has an S ability but no hidden one.

### CODE REVIEW

**Comment by asgdf** [comment]
(bump)

**Comment by Zarel** [comment]
My client fix makes it more visible the difference between an H and an S ability, just sayin'.

**Comment by Zarel** [comment]
To be fair, my client fix is uglier.

**Comment by Zarel** [comment]
Thoughts on that tradeoff?

**Comment by Zarel** [comment]
https://github.com/Zarel/Pokemon-Showdown-Client/commit/4bd843ee4cb8718307a3b075cf92b1704fa1cc66

is my fix, for the record.

**Comment by Zarel** [comment]
@mathfreak231 although, feel free to have an opinion on how this is best fixed.

My main problem is that in both your fixes, H and S abilities look identical. I think it should be possible to tell them apart.

**Comment by asgdf** [comment]
Your fix does make that clear, but only to people who already know about H and S abilities and how PS displays them. To everyone else, it might look like a bug ("Why are the abilities formatted so weirdly? It thinks there's two but there's only one"). Maybe with added tooltips or something.

**Comment by Zarel** [comment]
Hm, we could put the special ability in parens, so it'd be different from the Hidden Ability in italics?

**Comment by asgdf** [comment]
Parentheses would probably be confusing. In that context my instinct would be to read it as "kind of, but...", which is how it's used for "OU by technicality", but that doesn't really apply in this case.

I think it should either be a small enough change that it does not raise questions for most people (like just not italicizing it, or something), or it should fully explain what the distinction is, like with a tooltip.

Is there a reason to set H and S abilities apart at all, beyond flavor?

**Comment by Zarel** [comment]
"Kind of, but" is about as close as it gets, though. It's technically not the same forme, after all.

**Comment by asgdf** [comment]
Is it not? Isn't it just an ability that the regular forme can have that enables it to switch to a different forme?

**Comment by Zarel** [comment]
No, it's a different forme altogether, it just looks identical to the regular forme.

We list it as the same forme because there are no visual differences, so competitively your opponent can't tell the difference until you use the ability.

**Comment by KrisXV** [comment]
^ it's just like Rockruff and Rockruff-Dusk i believe

**Comment by asgdf** [comment]
Huh. Fair enough. The distinction will probably be lost on most players though I imagine. I still think we should explain it if we're going to make a point of it.

**Comment by Zarel** [comment]
Where exactly do you plan on fitting an explanation in a dex column?

I feel like the parens are clear enough. Implies something special, which is about the best we can do. Would an asterisk make you feel better?

**Comment by asgdf** [comment]
As I said, I think a tooltip or something of that sort might work.

It's just that when you put an asterisk or parentheses there, people will see that as "this is special, something is different with this one", but 99% of them won't know _how_ it differs, and they have no way of finding out without explicitly asking about it, and me and the rest of the Help room would be the ones answering those questions, presumably repeatedly.
The question most of them will probably have is "What does this mean? Can use it? Does it limit my set if I do?". The fact that it's technically a separate form that just looks the same is kind of a technicality of the implementation, and is by itself not very relevant to the average team-building user. If we do set it apart, then the reason should be a difference in mechanics in practise.
So, what impact does the specialness of these abilities have in practise? As I understand it, they are essentially event-only, in that they are incompatible with egg moves and place limitations on the DVs (and as such the HP types). So what I'm suggesting is putting an asterisk (or something) there, and tying it to a tooltip that says "Event only" or something similar. That phrases the difference in a way that most users who know at least some things about teambuilding mechanics will understand, as they will be familiar with the concept of event moves and the limitations these bring with them. I'm open to suggestions there, but that's the core idea I'm going for.

**Comment by Zarel** [comment]
We don't really have room for tooltips in a dex column. Maybe in the dex entry.

**Comment by asgdf** [comment]
But a mouse-over hint doesn't really take up room, does it?

**Comment by Zarel** [comment]
I mean, like, room as in room in the DOM tree, not visible room.

Browsers get slower the more complex the DOM tree is, and dex tables are already the most complex thing in PS and a major lag source.

**Comment by asgdf** [comment]
Fair enough I guess. My second choice would be to not mark S abilities in a special way, because if we can't explain it then I don't think it'll help anyone who didn't already know about it. Marking it with parens or an asterisk and nothing else would by my third choice. I'm still worried that we'd be creating more work for ourselves for minimal gain, but maybe I'm being too pessimistic about it.

**Comment by asgdf** [comment]
Well, I made my argument, it's your call from here. Clearly, the current version of showing "undefined" is the least optimal of all.

**Comment by Zarel** [comment]
You've yet to implement the compromise choice.

**Comment by Zarel** [comment]
I guess I will if no one does eventually, but I have a pretty big to-do list right now.

**Comment by asgdf** [comment]
I know, what I'm asking is what the compromise choice _is_.

**Comment by Zarel** [comment]
Parens.

**Comment by asgdf** [comment]
Okay. I still think it's kinda confusing but okay.
grafik
Like this?

File: server/chat.js [hunk 1]
```diff
@@ -1532,10 +1532,13 @@ Chat.getDataPokemonHTML = function (template, gen = 7, tier = '') {
 		} else {
 			buf += '<span class="col abilitycol">' + template.abilities['0'] + '</span>';
 		}
-		if (template.abilities['S']) {
-			buf += '<span class="col twoabilitycol' + (template.unreleasedHidden ? ' unreleasedhacol' : '') + '"><em>' + template.abilities['H'] + '<br />' + template.abilities['S'] + '</em></span>';
+		if (template.abilities['H'] && template.abilities['S']) {
+			buf += '<span class="col twoabilitycol' + (template.unreleasedHidden ? ' unreleasedhacol' : '') + '"><em>' + template.abilities['H'] + '</em><br />(' + template.abilities['S'] + ')</span>';
```
Comment (Zarel) [inline server/chat.js]:
```suggestion
			buf += '<span class="col twoabilitycol' + (template.unreleasedHidden ? ' unreleasedhacol' : '') + '"><em>' + template.abilities['H'] + '</em><br />(<em>' + template.abilities['S'] + '</em>)</span>';
```

File: server/chat.js [hunk 2]
```diff
@@ -1532,10 +1532,13 @@ Chat.getDataPokemonHTML = function (template, gen = 7, tier = '') {
...
 		} else if (template.abilities['H']) {
 			buf += '<span class="col abilitycol' + (template.unreleasedHidden ? ' unreleasedhacol' : '') + '"><em>' + template.abilities['H'] + '</em></span>';
+		} else if (template.abilities['S']) {
+			// special case for Zygarde
+			buf += '<span class="col abilitycol">(' + template.abilities['S'] + ')</span>';
```
Comment (Zarel) [inline server/chat.js]:
```suggestion
			buf += '<span class="col abilitycol">(<em>' + template.abilities['S'] + '</em>)</span>';
```

**Comment by asgdf** [comment]
I understood 
> Hm, we could put the special ability in parens, so it'd be different from the Hidden Ability in italics?

as not putting the parenthesized ability in italics.


grafik
The text overlap on Zygarde bothers me quite a bit

**Comment by Zarel** [comment]
I don't like it either, but do you have a better idea?

**Comment by Zarel** [comment]
Overline?

**Comment by Zarel** [comment]
Asterisk?

**Comment by Zarel** [comment]
Dotted underline?

**Comment by asgdf** [comment]
As I said, I don't think we shoukd be highlighting the ability in a different way if we can't explain to users what the difference is. 
A tooltip or hover text would be ideal, but since that's not possible and if we really want to format it differently, then parens are the least bad option I guess. Asterisks and dotted underlines say "there's more information here! Mouse over" to me, so that would be misleading.

**Pull request merged**